### PR TITLE
feature/CU-jv46zq_FRONT-END-Create-quiz-wizard

### PR DESCRIPTION
**Pull request by Thurstag**

### CODE REVIEW

**Comment by Thurstag** [comment]
Task linked: [CU-jv46zq [FRONT-END] Create quiz wizard](https://app.clickup.com/t/jv46zq)

**Comment by Thurstag** [comment]
image

File: lib/model/api/quiz.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  /// Load a question from a JSON
+  static AQuizQuestion fromJSON(Map<String, dynamic> json) {
+    return StringMultipleChoiceQuestion.fromJSON(json);
+  }
+}
+
+/// Question with multiple choices defined by a string
+class StringMultipleChoiceQuestion extends AQuizQuestion {
+  static const String QUESTION_FIELD_NAME = 'question';
+  static const String ANSWERS_FIELD_NAME = 'answers';
```
Comment (Thurstag) [inline lib/model/api/quiz.dart]: `static const String CHOICES_FIELD_NAME = 'choices';`

File: lib/model/api/quiz.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  final String question;
+
+  /// Constructor
+  StringMultipleChoiceQuestion(String question, List<String> answers)
+      : question = question,
+        answers = answers.toList();
+
+  /// Load a StringMultipleChoiceQuestion from a JSON
+  factory StringMultipleChoiceQuestion.fromJSON(Map<String, dynamic> json) {
+    if (!json.containsKey(QUESTION_FIELD_NAME) ||
```
Comment (Thurstag) [inline lib/model/api/quiz.dart:50]:
```
    if (!json.containsKey(QUESTION_FIELD_NAME) ||
        !json.containsKey(CHOICES_FIELD_NAME) ||
        !(json[QUESTION_FIELD_NAME] is String) ||
        !(json[CHOICES_FIELD_NAME] is List<dynamic>) ||
        (json[CHOICES_FIELD_NAME] as List<dynamic>)
            .any((element) => !(element is String))) {
      throw Exception('Malformed StringMultipleChoiceQuestion');
    }

    return StringMultipleChoiceQuestion(
        json[QUESTION_FIELD_NAME],
        (json[CHOICES_FIELD_NAME] as List<dynamic>)
            .map((e) => e as String)
            .toList());
```
\+ Enrichir les tests

File: lib/widgets/quiz_form.dart [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+/// Quiz form data
+class QuizFormData {
+  final List<Object?> answers;
+
+  /// Constructor
+  QuizFormData(List<AQuizQuestion> questions)
+      : answers = List.generate(questions.length, (index) => null);
+}
```
Comment (Thurstag) [inline lib/widgets/quiz_form.dart:33]: Ajouter
```
/// Convert object into a JSON
  Map<String, dynamic> toJson() {
    return {'answers': answers};
  }
```

**Review by LeBoyLeoz — APPROVED** [review]

**Review by Fantorem — APPROVED** [review]

**Review by Tomansion — APPROVED** [review]

**Review by YesLife — APPROVED** [review]

**Pull request merged**